en/peripherals/frsky_telemetry.md

```diff
@@ -1,27 +1,65 @@
 
-## Pixracer
+## Flight Controller Hardware
 
-> **Success** FrSky telemetry auto-starts on Pixracer and
-  detects D or S mode, there is no need to configure anything.
+Pixhawk/PX4 supports D (old) and S (new) FrSky telemetry. FMUv4 based boards (Pixracer) include the required electronics for the interface, older boards (Pixhawk) need a UART to S.PORT adapter board.
 
-For receivers with D-port (D4R-II and similar): Connect the Pixracer
-FrSky TX line to the D4R-II RX line. Also connect GND.
+### Pixracer
 
-For receivers with S-port: Connect the Pixracer FrSky TX and RX lines
-together (solder the wires together) to the X series receiver's S.port
-pin. Also connect GND.
+#### Configuration
 
-## LuaPilot: Telemetry Screen on the Taranis
-If you have a Taranis that can receive the telemetry stream (e.g. a Taranis X9D
-Plus), you can install a script that shows you a nice overview of the received
-telemetry:
-![](../../images/taranis_telemetry.jpg)
+There is no need to configure anything in PX4 itself. Once connected (as discussed below) FrSky telemetry auto-starts on Pixracer and detects D or S mode. 
+
+#### Connecting Receivers with D-port
+
+Connect the Pixracer FrSky TX line to the receiver's RX line. Also connect GND (This applies to D4R-II and similar).
+
+![Radio Connection](../../assets/flight_controller/pixracer/grau_setup_pixracer_radio.jpg)
+
+
+#### Connecting Receivers with S-port
+
+Connect the Pixracer FrSky TX and RX lines together (solder the wires together) to the X series receiver's S.port pin. 
+Also connect GND. The S-port connection is shown below (using the provided I/O Connector).
+
+![Grau b Pixracer FrSkyS.Port Connection](../../assets/flight_controller/pixracer/grau_b_pixracer_frskys.port_connection.jpg)
+
+![Pixracer FrSkyS.Port Connection](../../assets/flight_controller/pixracer/pixracer_FrSkyTelemetry.png)
+
+
+
+### Pixhawk v2
+
+#### Configuration
+
+After connecting the board (see below), you also need to configure the port for use with FrSky, by setting the [SYS_COMPANION](../advanced_config/parameter_reference.md#SYS_COMPANION) parameter to 10 (see [Parameters](../advanced_config/parameters.md) for information on how).
+
+#### Connecting Receivers
+
+[Pixhawk 1](../flight_controller/pixhawk.md) (and [mRo Pixhawk](../flight_controller/mro_pixhawk.md) and other FMUv2 boards) connected to the receiver for FrSky telemetry via the TELEM2 UART. 
+
+You will need to connect via a UART to S.PORT adapter board, or using a [ready-made cable](#ready_made_cable).
+
+<!-- ideally add diagram here -->
+
+### Pixhawk Pro
+
+[Pixhawk 3 Pro](../flight_controller/pixhawk3_pro.md) can be connected to TELEM4 (no additional software configuration is needed).
+
+You will need to connect via a UART to S.PORT adapter board, or using a [ready-made cable](#ready_made_cable).
+
+
+## RC Transmitter Setup
+
+### Taranis - LuaPilot Setup

```

Comment:
I moved this down a level because we should one day allow extension to have other transmitters.